File: Contests/984/Quintomania.py
tests = int(input())

for i in range(tests):
    notes_count = int(input())

    notes = input().split(" ")

    perfect = True

    for j in range(notes_count - 1):
        semi_tones = abs(int(notes[j]) - int(notes[j+1]))
        if (semi_tones != 5 and semi_tones != 7):
            perfect = False
            break
    
    if perfect:
        print("YES")
    else:
        print("NO")

File: Contests/984/Quintomania.go
    package main
     
    import (
    	"bufio"
    	"fmt"
    	"math"
    	"os"
    )
     
    func main() {
    	reader := bufio.NewReader(os.Stdin)
    	writer := bufio.NewWriter(os.Stdout)
    	defer writer.Flush()
     
    	var n int
    	fmt.Fscan(reader, &n)
     
    	for i := 0; i < n; i++ {
    		var notesCount int
    		var notes []int
    		perfect := true
     
    		fmt.Fscan(reader, &notesCount)
     
    		for j := 0; j < notesCount; j++ {
    			var note int
    			fmt.Fscan(reader, &note)
    			notes = append(notes, note)
     
    			if j > 0 {
    				difference := math.Abs(float64(notes[j] - notes[j-1]))
    				if difference != 5 && difference != 7 {
    					perfect = false
    				}
    			}
    		}
     
    		if perfect {
    			fmt.Println("YES")
    		} else {
    			fmt.Println("NO")
    		}
    	}
    }